scylla/src/transport/topology.rs

```diff
@@ -534,23 +534,43 @@ async fn query_user_defined_types(
 async fn query_tables(
     conn: &Connection,
 ) -> Result<HashMap<String, HashMap<String, Table>>, QueryError> {
-    let mut tables_query = Query::new("select keyspace_name, table_name from system_schema.tables");
+    let mut result: HashMap<String, HashMap<String, Table>> = HashMap::new();
+    query_tables_with(
+        conn,
+        "select keyspace_name, table_name from system_schema.tables",
+        &mut result,
+    )
+    .await?;
+    query_tables_with(
+        conn,
+        "select keyspace_name, view_name from system_schema.views",
+        &mut result,
+    )
```

Comment:
Would this query return an error when running with Cassandra which has materialized views disabled?
Is the table not created in such case, or is it created but empty?
Maybe we could add this case to CI.

Reply:
No, the schema table is there for compatibility reasons, it will just be empty